pep-9999.rst

```diff
@@ -0,0 +1,217 @@
+
+
+The auth.toml file
+''''''''''''''''''
+
+Credentials are stored in the ``auth.toml`` file, under the ``[authentication]`` section. There is a correlation between the repository name and the authentication section's name::
+
+    [authentication]
+
+    [authentication.pypi]
+    username = "__token__"
+    password = "pypi-AgEIcHlwaS5vcmcCJDJjMmI5NDJmLWIzNzQtNDc3ZC04ZmRkLTM4NDEzY2M1ZTU5MAACJXsicGVybWlzc2lvbnMiOiAidXNlciIsICJ2ZXJzaW9uIjogMX0AAAYgV91r8iEUUz7Q5xbsfi_KGiO3wrJR2_IOOpnFyLo62bk"
+
+    [authentication.myrepo1]
+    username = "foo"
+    password = "YmFyCg=="
+
+
+Instead of specifying the credentials in this file, Keyring can be used::
+
+    keyring set https://pypi.org/simple/ __token__
+    keyring set <repository URL> <private-repository username>
+
+
+A repository's credentials has the following attributes::
+
+    ==================  ===========  ===============================================
+    Attribute           Type         Description
+    ==================  ===========  ===============================================
+    cert                str          Path to certificate authority bundle (*.pem)
+    cert-client         str          Path to client certificate (*.pem)
+    password            str          The password for the repository, base64-encoded. When token is used, enter it here in clear-text.
+    proxy_password      str          The password for the proxy server, base64-encoded.
+    proxy_username      str          The username for the proxy server
+    username            str          The username for the repository. When token is used, set this to ``__token__``.
+    ==================  ===========  ===============================================
+
+
+Package discovery
+-----------------
+
+The PyPI repository (``repositories.pypi``) does not have to be specified in the ``config.toml``. If not specified, it will be used as the first package index by default. However, by explicitly defining it, the repository attributes can be used and it can be ordered differently.
+
+The order of the repository definitions in ``config.toml`` shall be respected when searching for packages. This allows custom package indexes to take precendence over e.g. a default PyPI repository and the use of package shadowing.
+
+
+Configuration file discovery
+----------------------------
+
+The locations of the configuration files vary slightly across platforms. You may have per-user, per-virtualenv or global (shared amongst all users) configuration:
+
+If multiple configuration files are found, then they are combined (merge and override) in the following order:
+
+1. The global file is read
+2. The per-user file is read
+3. The virtualenv-specific file is read
+
+Each file read overrides any values read from previous files.
+
+
+Global
+''''''
+
+* Unix: ``$HOME/.config/python/``
+* macOS: ``$HOME/.config/python/``
+* Windows: ``%APPDATA%\python\``
```

Comment:
```suggestion
* Windows: ``%PROGRAMDATA%\Python``
```

Reply:
I believe `$HOME` isn't per-machine either for those other platforms. Presumably it belongs under `etc`?

Reply:
Oops, yes, that is right.

By the way, I chose `%APPDATA%` for Windows only because that's what pip is using for [pip.conf](https://pip.pypa.io/en/stable/user_guide/#searching-for-packages). But I don't know well enough what's the most suitable location here and we can go with `%PROGRAMDATA%` if you think that's more appropriate.